builtin/logical/ssh/path_creds_create.go

```diff
@@ -0,0 +1,195 @@
+	if ipAddr == nil {
+		return logical.ErrorResponse(fmt.Sprintf("Invalid IP '%s'", ipRaw)), nil
+	}
+	ip := ipAddr.String()
+	ipMatched, err := cidrContainsIP(ip, role.CIDR)
+	if err != nil {
+		return logical.ErrorResponse(fmt.Sprintf("Error validating IP: %s", err)), nil
+	}
+	if !ipMatched {
+		return logical.ErrorResponse(fmt.Sprintf("IP[%s] does not belong to role[%s]", ip, roleName)), nil
+	}
+
+	var result *logical.Response
+	if role.KeyType == KeyTypeOTP {
+		// Generate salted OTP
+		otp := uuid.GenerateUUID()
+		otpSalted := b.salt.SaltID(otp)
+		entry, err := req.Storage.Get("otp/" + otpSalted)
+		// Make sure that new OTP is not replacing an existing one
+		for err == nil && entry != nil {
+			otp := uuid.GenerateUUID()
+			otpSalted := b.salt.SaltID(otp)
+			entry, err = req.Storage.Get("otp/" + otpSalted)
+		}
+		entry, err = logical.StorageEntryJSON("otp/"+otpSalted, sshOTP{
+			Username: username,
+			IP:       ip,
+		})
+		if err != nil {
+			return nil, err
+		}
+		if err := req.Storage.Put(entry); err != nil {
+			return nil, err
+		}
+		result = b.Secret(SecretOTPType).Response(map[string]interface{}{
+			"key_type": role.KeyType,
+			"key":      otp,
+		}, map[string]interface{}{
+			"otp": otp,
+		})
+	} else if role.KeyType == KeyTypeDynamic {
+		// Fetch the host key to be used for dynamic key installation
+		keyEntry, err := req.Storage.Get(fmt.Sprintf("keys/%s", role.KeyName))
+		if err != nil {
+			return nil, fmt.Errorf("key '%s' not found error:%s", role.KeyName, err)
+		}
+
+		if keyEntry == nil {
+			return nil, fmt.Errorf("key '%s' not found", role.KeyName, err)
+		}
+
+		var hostKey sshHostKey
+		if err := keyEntry.DecodeJSON(&hostKey); err != nil {
+			return nil, fmt.Errorf("error reading the host key: %s", err)
+		}
+
+		// Generate RSA key pair
+		dynamicPublicKey, dynamicPrivateKey, err := generateRSAKeys()
+		if err != nil {
+			return nil, fmt.Errorf("error generating key: %s", err)
+		}
+
+		// Transfer the public key to target machine
+		err = uploadPublicKeyScp(dynamicPublicKey, username, ip, role.Port, hostKey.Key)
+		if err != nil {
+			return nil, err
+		}
+
+		// Add the public key to authorized_keys file in target machine
+		err = installPublicKeyInTarget(role.AdminUser, username, ip, role.Port, hostKey.Key)
+		if err != nil {
+			return nil, fmt.Errorf("error adding public key to authorized_keys file in target")
+		}
+
+		result = b.Secret(SecretDynamicKeyType).Response(map[string]interface{}{
+			"key":      dynamicPrivateKey,
+			"key_type": role.KeyType,
+		}, map[string]interface{}{
+			"admin_user":         role.AdminUser,
+			"username":           username,
+			"ip":                 ip,
+			"host_key_name":      role.KeyName,
+			"dynamic_public_key": dynamicPublicKey,
+			"port":               role.Port,
+		})
+	} else {
+		return nil, fmt.Errorf("key type unknown")
+	}
+
+	// Change the lease information to reflect user's choice
+	lease, _ := b.Lease(req.Storage)
+	if lease != nil {
```

Comment:
If the lease is nil, the default values in the secret will be used.
Only if the lease is not nil, it implies that user has used the /config/lease endpoint before creating credential.
Currently, both OTP and Dynamic secrets have duration as 10 minute and grace period as 2 minute.